File: spring_microservices/coffeeservice/src/main/java/dev/sotoestevez/coffeeservice/web/mappers/DateMapper.java
package dev.sotoestevez.coffeeservice.web.mappers;

import org.springframework.stereotype.Component;

import java.sql.Timestamp;
import java.time.OffsetDateTime;
import java.time.ZoneOffset;

@Component
public class DateMapper {

    public OffsetDateTime asOffsetDateTime(Timestamp time) {
        if (time == null) {
            return null;
        }
        return OffsetDateTime.of(time.toLocalDateTime(), ZoneOffset.UTC);
    }

    public Timestamp asTimestamp(OffsetDateTime time) {
        if (time == null) {
            return null;
        }
        return Timestamp.valueOf(time.atZoneSameInstant(ZoneOffset.UTC).toLocalDateTime());
    }

}

File: spring_microservices/orderservice/src/main/kotlin/dev/sotoestevez/orderservice/web/mappers/DateMapper.kt
package dev.sotoestevez.orderservice.web.mappers

import org.springframework.stereotype.Component
import java.sql.Timestamp
import java.time.OffsetDateTime
import java.time.ZoneOffset

@Component
class DateMapper {

    fun map(timestamp: Timestamp): OffsetDateTime {
        return OffsetDateTime.of(timestamp.toLocalDateTime(), ZoneOffset.UTC)
    }

    fun map(offsetDateTime: OffsetDateTime): Timestamp {
        return Timestamp.from(offsetDateTime.toInstant())
    }

}